### ISSUE TITLE
Specify a full BIDS model. Move transformations and contrasts to model. 

### ISSUE THREAD

**Issue by adelavega** [neuroscout/neuroscout#225]
This will actually be a fairly major one, since previously the assumption was that the backend would construct the model given the right fields. But now, we will construct the whole model on the frontend.

This is a good idea for various reasons, although its more work on the frontend. One of them is that it decouples the frontend from Neuroscout. Potentially the logic user here, if sufficiently modular, could be used in a similar app that helps users make models of local datasets.

There are a few things that need to get done:

- [ ] Move transformations and contrasts into the model
- [ ] List the names of the predictors selected in "HRF_variables" and "variables" (these are the names of the predictors selected by the user *prior* to any transformations.  These get lised at "level" = run and dataset.
- [ ] Fill in "inputs". The "inputs" field is the set of unique "entities" that identify the runs. I would come up with the list by inspecting the fields "number" "session" "subject" in the selected runs, and idetifying the unique set of these items. In addition "task" should be included in inputs as well.

**Comment by adelavega** [comment]
Here's an example of what a valid POST/PUT request without those fields (and the information in "model") looks like:

```
{
  "TR": 1.5,
  "compiled_at": null,
  "created_at": "2018-04-20T22:0",
  "data": {},
  "dataset_id": 5,
  "description": "",
  "hash_id": "dz",
  "model": {
    "blocks": [
      {
        "level": "run",
        "model": {
          "HRF_variables": [
            "brightness",
            "vibrance",
            "sharpness"
          ],
          "variables": [
            "brightness",
            "vibrance",
            "sharpness"
          ]
        },
        "transformations": [
          {
            "input": [
              "brightness",
              "vibrance",
              "sharpness"
            ],
            "name": "scale"
          }
        ]
      },
      {
        "level": "dataset",
        "model": {
          "variables": [
            "brightness",
            "vibrance",
            "sharpness"
          ]
        }
      }
    ],
    "description": "",
    "input": {
      "subject": [
        "24",
        "34"
      ],
      "task": "MerlinMovie"
    },
    "name": "MerlinCompareGOD"
  },
  "modified_at": "2018-04-20T22:0",
  "name": "MerlinCompareGOD",
  "parent_id": "V1",
  "predictions": "",
  "predictors": [
    {
      "id": 351
    },
    {
      "id": 352
    },
    {
      "id": 353
    }
  ],
  "private": true,
  "runs": [
    {
      "id": 134
    },
    {
      "id": 117
    }
  ],
  "status": "PASSED",
  "task_name": "MerlinMovie",
  "user_id": 1
}
```

**Comment by adelavega** [comment]
@tyarkoni I also realized that we haven't discussed how to deal with nuissance variables such as motion regressors. These regressors would be listed under "variables" a the first level, but not "HRF_variables". They also would not be listed at the second level. Currently, there's no way to specify this in the front end, although these regressors are ingested and available.

One shortcut with this for now would be just always include a standard "set" of nuisance variables in models by default. They could be hidden from the "Predictors" route, but fitlins should still find them for execution because they will be in the fmriprep folder.

**Comment by adelavega** [comment]
Closed w/ #245
